src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -0,0 +1,156 @@
+
+        with tempfile.NamedTemporaryFile(delete=False) as tf:
+            with open(lvs_deck, 'r') as f:
+                tf.write(pattern.sub("", f.read()).encode('utf-8'))
+            shutil.copystat(lvs_deck, tf.name)
+            shutil.copy(tf.name, lvs_deck)
+
+    def generate_multi_vt_gds(self) -> None:
+        """
+        vendor only provide SLVT gds, this patch will generate other 3(LVT, RVT, SRAM) VT gds file.
+        """
+        self.logger.info("generate gds for Multi-VT cells")
+
+        orig_gds = os.path.join(self.extracted_tarballs_dir, "ASAP7_PDKandLIB.tar/ASAP7_PDKandLIB_v1p5/asap7libs_24.tar.bz2/asap7libs_24/gds/asap7sc7p5t_24.gds")
+        # load original_gds
+        asap7_original_gds = gdspy.GdsLibrary(infile=orig_gds)
+        original_cells = asap7_original_gds.cell_dict
+        # WTF is this?
+        del original_cells['m1_template']
+        # required libs
+        multi_libs = {
+            "R": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": None,
+                },
+            "L": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": 98
+                },
+            "SL": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": 97
+                },
+            "SRAM": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": 110
+                },
+        }
+        # create new libs
+        for vt, multi_lib in multi_libs.items():
+            multi_lib["lib"].name = asap7_original_gds.name.replace('SL', vt)
+
+        for cell in original_cells.values():
+            poly_dict = cell.get_polygons(by_spec=True)
+            # extract polygon from layer 100(the boundary for cell)
+            boundary_polygon = poly_dict[(100, 0)]
+            for vt, multi_lib in multi_libs.items():
+                mvt_layer = multi_lib['mvt_layer']
+                if mvt_layer:
+                    # copy boundary_polygon to mvt_layer to mark the this cell is a mvt cell.
+                    mvt_polygon = gdspy.PolygonSet(boundary_polygon, multi_lib['mvt_layer'], 0)
+                    mvt_cell = cell.copy(name=cell.name.replace('SL', vt), exclude_from_current=True, deep_copy=True).add(mvt_polygon)
+                else:
+                    # RVT, just copy the cell
+                    mvt_cell = cell.copy(name=cell.name.replace('SL', vt), exclude_from_current=True, deep_copy=True)
+                # add mvt_cell to corresponding multi_lib
+                multi_lib['lib'].add(mvt_cell)
+
+        for vt, multi_lib in multi_libs.items():
+            # write multi_lib with 2.5e-10 precision
+            multi_lib['lib'].precision = 2.5e-10
+            multi_lib['lib'].write_gds(os.path.splitext(orig_gds)[0] + '_' + vt + '.gds')
+
+    def fix_sram_cdl_bug(self) -> None:
+        """
+        vendor's SRAM cdl use slvt cell, this patch will sed cells name in which, fix this bug.
```

Comment:
I don't understand this comment.

Reply:
"this patch will replace 'slvt' in SRAM cell names with 'sram' to indicate the correct VT for the SRAM cells"